### PR TITLE
Remove extra dependencies

### PR DESCRIPTION
**Pull request by dhananjay-patki**
Removed ROS 1 deps from constraintsolver and tracing, as well as removing an unused boost dependency we were not using

### CODE REVIEW

File: alica_solver_interface/CMakeLists.txt [hunk 1]
```diff
@@ -58,7 +58,8 @@ add_executable(${PROJECT_NAME}-tests
 
 target_link_libraries(${PROJECT_NAME}-tests PRIVATE Threads::Threads ${PROJECT_NAME} ${GTEST_LIBRARIES})
 
-add_test(NAME alica_solver_interface-tests COMMAND alica_solver_interface-tests)
+add_test(NAME IntervalTest.Double COMMAND ${PROJECT_NAME}-tests --gtest_filter=IntervalTest.Double --gtest_also_run_disabled_tests)
+add_test(NAME IntervalTest.Operations COMMAND ${PROJECT_NAME}-tests --gtest_filter=IntervalTest.Operations --gtest_also_run_disabled_tests)
```
Comment (Dale-Koenig) [inline alica_solver_interface/CMakeLists.txt]: No reason to run disabled tests I think

**Review by Dale-Koenig — APPROVED** [review]

File: supplementary/alica_tracing/CMakeLists.txt [hunk 2]
```diff
@@ -2,26 +2,16 @@ cmake_minimum_required(VERSION 3.5.1)
...
-  INCLUDE_DIRS include
-  CATKIN_DEPENDS  roscpp ${PKG_DEPS}
-  LIBRARIES ${PROJECT_NAME} jaegertracing
-  DEPENDS alica_engine
-)
 
 include_directories(
   include
-  ${catkin_INCLUDE_DIRS}
   ${alica_engine_INCLUDE_DIRS}
```
Comment (triccyx) [inline supplementary/alica_tracing/CMakeLists.txt:15]: can be removed

File: supplementary/constraintsolver/CMakeLists.txt [hunk 3]
```diff
@@ -3,36 +3,30 @@ project(constraintsolver)
...
- LIBRARIES constraintsolver
- CATKIN_DEPENDS   roslib 
- DEPENDS fsystem alica_common_config alica_solver_interface autodiff alica_engine
-)
-
 include_directories(
   include
-  ${catkin_INCLUDE_DIRS}
   ${alica_common_config_INCLUDE_DIRS}
   ${alica_engine_INCLUDE_DIRS}
```
Comment (triccyx) [inline supplementary/constraintsolver/CMakeLists.txt:25]: can be removed

File: supplementary/constraintsolver/test/test_cnsat.cpp [hunk 4]
```diff
@@ -11,8 +11,8 @@
 #include <fstream>
 #include <iostream>
 #include <memory>
-#include <ros/package.h>
-#include <ros/ros.h>
+// #include <ros/package.h>
```
Comment (triccyx) [inline supplementary/constraintsolver/test/test_cnsat.cpp]: Can be removed?

File: supplementary/constraintsolver/test/test_constraintsolver.cpp [hunk 5]
```diff
@@ -9,8 +9,8 @@
 #include <engine/AlicaClock.h>
 #include <yaml-cpp/yaml.h>
 
-#include <ros/package.h>
-#include <ros/ros.h>
+// #include <ros/package.h>
```
Comment (triccyx) [inline supplementary/constraintsolver/test/test_constraintsolver.cpp]: Can be removed?

**Comment by bjoernschroeder** [comment]
Both in CI and locally when running tests with catkin run_tests we get this error message

```
Exception in callback _UnixReadPipeTransport._read_ready()
handle: <Handle _UnixReadPipeTransport._read_ready()>
Traceback (most recent call last):
  File "/usr/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.8/asyncio/unix_events.py", line 500, in _read_ready
    self._protocol.data_received(data)
  File "/usr/lib/python3/dist-packages/osrf_pycommon/process_utils/async_execute_process_asyncio/impl.py", line 109, in data_received
    protocol.on_stderr_received(data)
  File "/usr/lib/python3/dist-packages/catkin_tools/execution/io.py", line 269, in on_stderr_received
    self.log_file.write(data)
ValueError: write to closed file
```

Tests and CI are still passing but is there any way to get rid of that?

**Comment by dhananjay-patki** [comment]
Interesting that the error doesn't show up while building/testing on colcon. I'll see what's up with that error

**Comment by dhananjay-patki** [comment]
> Interesting that the error doesn't show up while building/testing on colcon. I'll see what's up with that error

When running each package individually with `catkin test --this` I don't get the write to closed file error. No idea what's causing that error or how to fix it, other than maybe switching over to colcon.

File: supplementary/alica_tracing/CMakeLists.txt (hunk 2 above)
Comment (dhananjay-patki) [reply supplementary/alica_tracing/CMakeLists.txt:15]: Done

File: supplementary/constraintsolver/CMakeLists.txt (hunk 3 above)
Comment (dhananjay-patki) [reply supplementary/constraintsolver/CMakeLists.txt:25]: Done

File: supplementary/constraintsolver/test/test_cnsat.cpp (hunk 4 above)
Comment (dhananjay-patki) [reply supplementary/constraintsolver/test/test_cnsat.cpp]: Done!

File: supplementary/constraintsolver/test/test_constraintsolver.cpp (hunk 5 above)
Comment (dhananjay-patki) [reply supplementary/constraintsolver/test/test_constraintsolver.cpp]: Done!

**Pull request closed without merging**